client/src/pages/FAQ/FAQ.jsx

```diff
@@ -1,9 +1,471 @@
-import React from 'react';
+import { React, useState } from 'react';
 import PropTypes from 'prop-types';
-import { getInformation } from './functions';
+import { getQuestions, getCategories, submitQuestion } from './functions';
+import './FAQ.scss';
+import Wave from '../../assets/misc/wave-reverse.png';
+import { ButtonSelector } from '../../components/buttonSelector/buttonSelector/ButtonSelector';
+import { SingleAccordion } from '../../components/text/Accordion/SingleAccordion/SingleAccordion';
+import { Button } from '../../components/button/Button/Button';
+import { TextInput } from '../../components/input/TextInput/TextInput';
+import SearchIcon from '../../assets/misc/magnifying-glass-solid.svg';
+import DeleteIcon from '../../assets/misc/circle-xmark-solid.svg';
 
 const PageFAQ = () => {
-  return <div style={{ backgroundColor: 'blue', height: '10vh' }}>{getInformation()}</div>;
+  const [activeIndex, setActiveIndex] = useState(0);
+  const [isSearch, setIsSearch] = useState(false);
+  const [isMultiSearch, setIsMultiSearch] = useState(false);
+  const [isNoMatch, setIsNoMatch] = useState(false);
+  const [selectedSearchResult, setSelectedSearchResult] = useState(false);
+  const [selectedQuestion, setSelectedQuestion] = useState(0);
+  const [selectedQuestions, setSelectedQuestions] = useState([]);
+  const [searchQuery, setSearchQuery] = useState('');
+  const data = getQuestions();
+  const unsortedQuestions = [];
+  const generalQuestions = [];
+  const FroshKitsQuestions = [];
+  const FroshGroupQuestions = [];
+  for (let i = 0; i < data.length; i++) {
+    if (data[i].category == 'General') {
+      generalQuestions.push({ question: data[i].question, answer: data[i].answer });
+    }
+    if (data[i].category == 'F!rosh Kits') {
+      FroshKitsQuestions.push({ question: data[i].question, answer: data[i].answer });
+    }
+    if (data[i].category == 'F!rosh Group') {
+      FroshGroupQuestions.push({ question: data[i].question, answer: data[i].answer });
+    }
```

Comment:
Is there a reason why these are hardcoded? It seems strange to hard code three categories explicitly

Reply:
Resolved by using an object and properties to dynamically add new categories based on the categories set in the data.